packages/core/src/sdk/UserInfoSingleton/types.ts

```diff
@@ -5,8 +5,11 @@
  */
 
 export type UserInfo = {
-    readonly id?: string;
+    readonly id?: string /** @deprecated To be made mandatory when removing DdSdkReactnative.setUser */;
     readonly name?: string;
     readonly email?: string;
-    readonly [key: string]: unknown;
+    readonly extraInfo?: Record<string, unknown>;
+    readonly [
+        key: string
+    ]: unknown /** @deprecated To be removed alongside DdSdkReactnative.setUser */;
```

Comment:
This should be removed once setUser is deprecated as it "breaks" the parameter structure and contract with its native counterparts.